frontend/src/components/object/ObjectTable.vue

```diff
@@ -98,6 +131,50 @@ const filters = ref({
   // @ts-ignore
   global: { value: null, matchMode: FilterMatchMode.CONTAINS }
 });
+
+const onSelectAllChange = (event: any) => {
+  selectAll.value = event.checked;
+  if (event.checked) {
+    if (isInitialLoad.value) {
+      /*
+       * On initial page load, the object order returned by this emit doesn't corespond
+       * to the actual display order - it's just the order returned by the COMS API.
+       * Workaround: directly query the DOM to figure out the actual display order (see queryTableOrder()).
```

Comment:
Isn't it the datatable component's job to be managing the order? I'm not completely sure why we need to be doing this, unless we need the explicit data ordering value in this parent component and there isn't any other way to pull that information from the datatable API?

Reply:
Initial page load is an edge case - typically, the event returned by the `value-change` emit is in sorted order, but only on page load does it return the order from the COMS API instead of what's displayed. If you paginate, change sort order etc after this initial load it'll be in the correct order though.